rscripts/YAL5-codon-pairs.R

```diff
@@ -0,0 +1,1455 @@
+#'                             hd_file = here::here("Mok-simYAL5",
+#'                                                   "output",
+#'                                                   "A",
+#'                                                   "A.h5"),
+#'                             gff_df,
+#'                             min_read_length = 10,
+#'                             snapdisp = 0L,
+#'                             asite_disp_path = here::here(
+#'                               "data",
+#'                               "yeast_standard_asite_disp_length.txt"),
+#'                             filter_for_frame = TRUE,
+#'                             feature_of_interest = "TCC AAG",
+#'                             expand_width = 5L)
+#'
+#' @export
+ExpandedRegionNormalisation <- function(codon_pos_pair_i200,
+                                        gene_names, dataset, hd_file,
+                                        gff_df, min_read_length, snapdisp,
+                                        asite_disp_path,
+                                        filter_for_frame,
+                                        feature_of_interest,
+                                        expand_width) {
+
+  # fetch the expanded tibbles for each occurrence of the feature of interest
+  expand_feature_region <- ExpandFeatureRegion(codon_pos_pair_i200,
+                                               gene_names,
+                                               dataset,
+                                               hd_file,
+                                               gff_df,
+                                               min_read_length,
+                                               snapdisp,
+                                               asite_disp_path,
+                                               filter_for_frame,
+                                               feature_of_interest,
+                                               expand_width)
+
+  # function to normalise each tibble before they can be overlayed
+  Normalization <- function(.x, expand_width) {
+    dplyr::mutate(.x, RelCount = Count / sum(Count) * (2 * expand_width + 1))
+ }
+
+  normalized_expand_list <- purrr::map(.x = expand_feature_region,
+                                       .f = Normalization,
+                                       expand_width)
+
+  # function to remove the occurrences of tibbles which contain NaN
+  CheckForNaN <- function(.x) {
+
+    normalized_expand_list <- .x
+
+    Relcount_values <- unlist(normalized_expand_list$RelCount)
+
+    print(Relcount_values)
+
+    SetNaNToZero <- function(Relcount_values) {
+
+      if (is.nan(Relcount_values)) {
+        print("NaN present")
+        Relcount_values <- 0
+      } else {
+        Relcount_values <- Relcount_values
+      }
+    }
+    Relcount_values <- unlist(purrr::map(.x = Relcount_values,
+                                         .f = SetNaNToZero))
+
+    print(Relcount_values)
```

Comment:
remove from final code, good for testing but makes the console cluttered for user